File: book_exercises/chapter_3/functions/src/main.c
#include <stdio.h>

void raise_to_power(int x, int power) {
    size_t result = 1;

    for (int i = 1; i <= power; i++) {
        result *= x; 
    }
    printf("%d to the power of %d: %zu\n", x, power, result);
}

int main() {
    printf("Raise number to power.\n");

    int number, power;
    printf("Enter number: ");
    scanf("%d", &number);
    printf("\nEnter power: ");
    scanf("%d", &power);

    raise_to_power(number, power);

    return 0;
}

File: book_exercises/chapter_3/functions/src/main.rs
use std::io;

fn raise_to_power(x: usize, power: i32) {
    let mut result: usize = 1;
    for _ in 1..(power + 1) {
       result = result * x; 
    }
    println!("{x} to the power of {power}: {result}");
}

fn five() -> i32 {
    5
}

fn plus_one(x: i32) -> i32 {
    x + 1
}

fn main() {
    println!("Raise number to the power.");

    let mut number = String::new();
    let mut power = String::new();

    println!("Enter the number:");
    io::stdin() 
        .read_line(&mut number)
        .expect("Failed to read.");
    let number = number
        .trim()
        .parse()
        .expect("Couldn't convert to integer.");

    println!("Enter the power:");
    io::stdin()    
        .read_line(&mut power)
        .expect("Failed to read.");
    let power = power
        .trim()
        .parse()
        .expect("Couldn't convert to integer.");

    raise_to_power(number, power);
    
    let x = plus_one(five());

    println!("X is {x}");
}
